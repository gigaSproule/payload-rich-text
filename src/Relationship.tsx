import type { Options } from "./types";

export interface RelationshipData {
  type: "relationship";
  format: string;
  version: number;
  relationTo: string;
  value: {
    id: string;
    [key: string]: unknown;
  };
}

export type Props = { data: RelationshipData; options?: Options };

export const Relationship = ({ data, options }: Props) => {
  return options?.relationship ? options.relationship(data) : undefined;
};
