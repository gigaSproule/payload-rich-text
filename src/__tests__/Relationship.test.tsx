import { describe, expect, it } from "@jest/globals";
import { render } from "@testing-library/react";
import type { Props } from "../Relationship";
import { Relationship } from "../Relationship";
import { createRelationshipData } from "./helpers/createRelationshipData";

describe("Relationship", () => {
  it("should not render anything if no options provided", () => {
    const relationshipData: Props["data"] = createRelationshipData();
    const { container } = render(<Relationship data={relationshipData} />);
    expect(container).toMatchSnapshot();
  });

  it("should not render anything if options provided without relationship", () => {
    const relationshipData: Props["data"] = createRelationshipData();
    const { container } = render(
      <Relationship data={relationshipData} options={{}} />,
    );
    expect(container).toMatchSnapshot();
  });

  it("should render custom render if options provided with relationship", () => {
    const relationshipData: Props["data"] = createRelationshipData();
    const { container } = render(
      <Relationship
        data={relationshipData}
        options={{
          relationship: (relationship) => {
            return (
              <div>Custom Relationship - {JSON.stringify(relationship)}</div>
            );
          },
        }}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
