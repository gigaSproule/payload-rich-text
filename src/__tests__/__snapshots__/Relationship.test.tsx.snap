// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Relationship should not render anything if no options provided 1`] = `<div />`;

exports[`Relationship should not render anything if options provided without relationship 1`] = `<div />`;

exports[`Relationship should render custom render if options provided with relationship 1`] = `
<div>
  <div>
    Custom Relationship - 
    {"type":"relationship","format":"","version":2,"relationTo":"page","value":{"id":"67922e89bff17903c13f53f0","some":"custom","block":"fields"}}
  </div>
</div>
`;
